dotCMS/src/main/java/com/dotcms/filters/interceptor/saml/SamlWebInterceptor.java

```diff
@@ -0,0 +1,544 @@
+            user       = this.userAPI.loadUserById(samlUserId, systemUser, false);
+        } catch (NoSuchUserException e) {
+
+            Logger.error(this, "No user matches ID '" + samlUserId + "'. Creating one...");
+            user = null;
+        } catch (Exception e) {
+
+            Logger.error(this, "An error occurred when loading user with ID '" + samlUserId + "'", e);
+            user = null;
+        }
+
+        return user;
+    }
+
+    /**
+     * Get's the include urls to be analyzed by the open saml plugin, usually
+     * the admin They can be a pattern
+     *
+     * @param identityProviderConfiguration {@link IdentityProviderConfiguration}
+     * @return String []
+     */
+    public String[] getIncludePathArray(final IdentityProviderConfiguration identityProviderConfiguration) {
+
+        final String accessFilterValues = this.samlConfigurationService.getConfigAsString(identityProviderConfiguration,
+                SamlName.DOT_SAML_INCLUDE_PATH_VALUES);
+
+        return UtilMethods.isSet(accessFilterValues)? accessFilterValues.split( "," ) : null;
+    }
+
+    /**
+     * Returns the logout paths
+     *
+     * @param identityProviderConfiguration {@link IdentityProviderConfiguration}
+     * @return String[]
+     */
+    public String[] getLogoutPathArray(final IdentityProviderConfiguration identityProviderConfiguration) {
+
+        final String logoutPathValues = this.samlConfigurationService.getConfigAsString(identityProviderConfiguration,
+                SamlName.DOT_SAML_LOGOUT_PATH_VALUES);
+
+        return UtilMethods.isSet(logoutPathValues)? logoutPathValues.split( "," ) : null;
+    }
+
+    /**
+     * Determine if the path is Backend Admin, usually it is for /c && /admin or
+     * if the path is a file or path, will check if the user has permission
+     *
+     * @param uri
+     *            {@link String}
+     * @param includePaths
+     *            {@link String} array
+     * @return boolean
+     */
+    protected boolean checkIncludePath(final String uri, final String... includePaths) {
+
+        boolean include = false;
+
+        // this is the backend uri test.
+        for (final String includePath : includePaths) {
+
+            Logger.debug(this, ()-> "Evaluating URI '" + uri + "' with pattern: " + includePath);
+
+            include |= RegEX.contains(uri, includePath);
+        }
+
+        Logger.debug(this, "Incoming URI '" + uri + "', include? " + include);
+
+        return include;
+    }
+
+    /**
+     * This method checks if some path does not wants to be treatment by the
+     * {@link SamlWebInterceptor} An example of exception might be the destroy.jsp, so
+     * on.
+     *
+     * @param uri
+     *            {@link String}
+     * @param filterPaths
+     *            {@link String} array
+     * @return boolean
+     */
+    protected boolean checkAccessFilters(final String uri, final String[] filterPaths) {
```

Comment:
Codacy found an issue: [Consider using varargs for methods or constructors which take an array the last parameter.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5707914)